scapy/sendrecv.py

```diff
@@ -74,25 +74,72 @@ class debug:
 #  Send / Receive  #
 ####################
 
-_DOC_SNDRCV_PARAMS = """
+_DOC_SNDRCV_PARAMS_HEAD = """
     :param pks: SuperSocket instance to send/receive packets
-    :param pkt: the packet to send
-    :param timeout: how much time to wait after the last packet has been sent
-    :param inter: delay between two packets during sending
-    :param verbose: set verbosity level
-    :param chainCC: if True, KeyboardInterrupts will be forwarded
-    :param retry: if positive, how many times to resend unanswered packets
-        if negative, how many times to retry when no more packets
-        are answered
-    :param multi: whether to accept multiple answers for the same stimulus
-    :param rcv_pks: if set, will be used instead of pks to receive packets.
-        packets will still be sent through pks
-    :param prebuild: pre-build the packets before starting to send them.
-        Automatically enabled when a generator is passed as the packet
-    :param _flood:
-    :param threaded: if True, packets will be sent in an individual thread
-    :param session: a flow decoder used to handle stream of packets
-    :param chainEX: if True, exceptions during send will be forwarded
+    :type pkt: SuperSocket
+    """
+
+_DOC_SNDRCV_PARAMS_BODY = """
+    :param pkt: Packet or iterable of packets to be sent.
+    :type pkt: _PacketIterable
```

Comment:
Actually, the types make it unreadable. Could you drop them?

Reply:
No problem. I'll check it out. I just added types, so that sphinx is able to show them. I mainly checked the api documentation.

Reply:
IDEs like the type parameter aswell... would it be an option for you, to change the order of the type information?

Example:
NOW:

```python
    """Forward traffic between interfaces if1 and if2, sniff and return
    the exchanged packets.

    :param if1: The interfaces to use (interface names or opened sockets).
    :type if1: _GlobInterfaceType

    :param if2: The interfaces to use (interface names or opened sockets).
    :type if2: _GlobInterfaceType

    :param xfrm12: A function to call when forwarding a packet from if1 to
        if2. If it returns True, the packet is forwarded as it. If it
        returns False or None, the packet is discarded. If it returns a
        packet, this packet is forwarded instead of the original packet.
    :type xfrm12: Optional[Callable[[Packet], Union[Packet, bool]]]

    :param xfrm21: Same as xfrm12 for packets forwarded from if2 to if1.
    :type xfrm21: Optional[Callable[[Packet], Union[Packet, bool]]]

    """
```

```python
    """Forward traffic between interfaces if1 and if2, sniff and return
    the exchanged packets.

    :param if1: The interfaces to use (interface names or opened sockets).
    :param if2: The interfaces to use (interface names or opened sockets).
    :param xfrm12: A function to call when forwarding a packet from if1 to
        if2. If it returns True, the packet is forwarded as it. If it
        returns False or None, the packet is discarded. If it returns a
        packet, this packet is forwarded instead of the original packet.
    :param xfrm21: Same as xfrm12 for packets forwarded from if2 to if1.
   
    :type if1: _GlobInterfaceType
    :type if2: _GlobInterfaceType
    :type xfrm12: Optional[Callable[[Packet], Union[Packet, bool]]]
    :type xfrm21: Optional[Callable[[Packet], Union[Packet, bool]]]

    """
    ```
PROPOSAL:

Reply:
Just to be sure, this shows as '_GlobInterfaceType' in the IDE? If that's the case I'd argue that we don't care. If it's smarter than that, then there's a case to be made.